### PR TITLE
Feature log category settings

### PR DESCRIPTION
**Pull request by tgxworld**
Meta: https://meta.discourse.org/t/category-security-changes-arent-logged/26820

### CODE REVIEW

File: app/assets/javascripts/discourse/controllers/edit-category.js.es6 [hunk 1]
```diff
@@ -67,17 +68,13 @@ export default Ember.Controller.extend(ModalFunctionality, {
       this.set('saving', true);
       model.set('parentCategory', parentCategory);
 
-      self.set('saving', false);
       this.get('model').save().then(function(result) {
+        self.set('saving', false);
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/controllers/edit-category.js.es6]: :+1:

File: app/assets/javascripts/discourse/controllers/edit-category.js.es6 [hunk 2]
```diff
@@ -67,17 +68,13 @@ export default Ember.Controller.extend(ModalFunctionality, {
...
         self.send('closeModal');
         model.setProperties({slug: result.category.slug, id: result.category.id });
         DiscourseURL.redirectTo("/c/" + Discourse.Category.slugFor(model));
       }).catch(function(error) {
-        if (error && error.responseText) {
-          self.flash($.parseJSON(error.responseText).errors[0], 'error');
-        } else {
-          self.flash(I18n.t('generic_error'), 'error');
-        }
+        self.flash(extractError(error), 'error');
```
Comment (ZogStriP) [inline app/assets/javascripts/discourse/controllers/edit-category.js.es6]: :heart_eyes:

File: app/controllers/categories_controller.rb [hunk 3]
```diff
@@ -175,4 +183,8 @@ def category_params
     def fetch_category
       @category = Category.find_by(slug: params[:id]) || Category.find_by(id: params[:id].to_i)
     end
+
+    def initialize_staff_action_logger
```
Comment (ZogStriP) [inline app/controllers/categories_controller.rb]: Do you really need this to be in a `before_filter` callback when it's only being used in 1 method?

Comment (tgxworld) [inline app/controllers/categories_controller.rb]: Hehe I was about to ask on meta whether we need to log categories that are deleted

Comment (ZogStriP) [inline app/controllers/categories_controller.rb]: Sure. Unless it's a PITA to support ;)

Comment (tgxworld) [inline app/controllers/categories_controller.rb]: https://meta.discourse.org/t/category-security-changes-arent-logged/26820/5

Ahh ok I'll need to support it

**Comment by tgxworld** [comment]
Updated :smile:

**Comment by ZogStriP** [comment]
Thanks :+1:

**Pull request merged**